### ISSUE TITLE
JetAppointment's calendar is not accessible for keyboard navigation and is practically invisible to screen readers

### ISSUE THREAD

**Issue by Tchelet-Levi** [CrocoBlock/suggestions#7544]
Hi, I apologize if this is not the correct way to bring this up, however the JetAppointment plugin's calendar seems to have a few accessibility issues with it that I would like to point out.

The most substantial one is that **it is currently not accessible via keyboard navigation**.
This means that anyone with some kind of disability that prevents them from using a mouse such as having Parkinson's would be unable to complete the form that it is attached to.

In addition to that, the HTML is not very descriptive for screen readers either. 
For example, using div elements instead of buttons for the time slots and dates, and a lack of aria-labels.

I'm not an accessibility expert by no means, but perhaps it is worth re-evaluating the calendar's structure to make sure that it is more accessible in the future.

Either way, I love the plugin and appreciate the work the team has been putting into it.
Thanks!

**Comment by Tchelet-Levi** [comment]
Hi everyone, I thought I would give a more in depth explanation of the problem/s and my proposed code solution.

### Please look at the video below and notice the following problems: 


https://github.com/user-attachments/assets/5278e1fb-4cbe-4a87-88d1-a7203e1f04b9



- The buttons to change the month do not have aria-labels, which means they will simply be read by a screen reader as "button" which is not descriptive enough.

- The user cannot tab into the calendar, meaning the entire form cannot be submitted by users who rely on navigating websites with their keyboard alone.

- The screen reader doesn't recognize this region as a calendar. There are no hints to a visually impaired user that there are any actions they can take (unlike the name field above it).

- When attempting to read the dates of the calendar, only the numbers will be read, which without context such as whether the selection is checked, or whether the button is disabled or not, does not provide enough information.

- Even when selecting the date with a mouse, there is no indication of what was selected or if selection even took place.

- Time slots are also not reachable with the keyboard.

- The close button is also not reachable with the keyboard. It also reads as "times" as it lacks an aria-label attribute and uses the ```&times;``` HTML entity as an icon.

- The time picker is reachable via keyboard navigation but is not very descriptive of its purpose.

- The recurring dates do not read the full name of the week day when selected.

There are a few more issues that I have noticed when using the other time slot methods (recurring, range, etc) but the idea is the same.

### My solution


https://github.com/user-attachments/assets/eb091025-108a-4e0d-9475-8de6f926bfe9



- Even though it is the recommended approach, I have opted into using ```aria-*``` attributes instead of changing the HTML markup from ```<divs/>``` and ```<spans/>``` to more semantic elements such as ```<h2/>```, etc in order to be as backwards compatible as possible to existing calendars. This also means that the preferred way of making an accessible calendar using a table structure is not possible due to the calendar using flex-wrap to create the grid-like structure as opposed to a tabular markup like the one with the ```<table/>``` element, but it's something we can live with ( [See the 'grid' role and its requirements](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/grid_role) ).

- Added roles, aria labels and descriptions where necessary.

- Made elements tab-able using tabindex, and added custom keyboard navigation functionality following the [Radio Group Pattern](https://www.w3.org/WAI/ARIA/apg/patterns/radio/examples/radio/) from w3.org.

- Supports custom week order.

#### Notes

- Made sure only to use features over 91% acceptance rate by modern browsers according to [caniuse.com.](https://caniuse.com/)

- Added documentation where I thought was necessary.

- Tested on my own to make sure it works with most variations in booking types and layouts, though since I do not have access to unit tests, more tests would be necessary.

Since there is no public repository for jet-appointment-booking (that I could find), I git init-ed one from the version I had at the time (2.1.0) and made my changes there on one new commit, so that it would be easier to compare the changes.

While the source code for the jet-appointment-booking plugin falls under GPL, I want to respect the Crocoblock team and not publish the entire thing without permission, possibly letting people profit off your hard work, so instead please contact me and I can send you a link to a private repo if you are interested in implementing it in the plugin officially.

And of course, please feel free to use the changes I wrote as you wish. I simply wish for this feature to be implemented as I find the plugin to be very useful but I don't feel comfortable giving my clients inaccessible forms when there are so many frivolous lawsuits surrounding web accessibility in my country and in the U.S (not sure about Europe).
It also makes sense for people with disabilities -- whether it is vision impairment or motor impairment -- to set an appointment using JetAppointment, and at the moment it is simply not possible for them.

Please let me know if there are any question!

**Comment by aalexandr21** [comment]
was added and released in JetAppointments 2.3.0
